### PR TITLE
fix(tree): check for all synonyms of conflicting default args

### PR DESCRIPTION
**Pull request by corneliusroemer**
- **fix(tree): check for all synonyms of conflicting args**
- **chore(tests): more robust test by adding `-redo` to iqtree args**

## Description of proposed changes

See commit messages for details.

## Related issue(s)

Inspired by https://github.com/iqtree/iqtree2/issues/273

## Checklist

- [x] Automated checks pass
- [x] [Check][1] if you need to add a changelog message
- [x] [Check][2] if you need to add tests
- [x] [Check][3] if you need to update docs

[1]: https://github.com/nextstrain/augur/blob/-/docs/contribute/DEV_DOCS.md#updating-the-changelog
[2]: https://github.com/nextstrain/augur/blob/-/docs/contribute/DEV_DOCS.md#testing
[3]: https://github.com/nextstrain/augur/blob/-/docs/contribute/DEV_DOCS.md#when-to-update

### CODE REVIEW

**Comment by corneliusroemer** [comment]
Would be nice if someone could look at this, e.g. @huddlej @victorlin @tsibley @genehack @joverlee521 @jameshadfield

File: augur/tree.py
Comment (genehack) [inline augur/tree.py:264]: _not at all familiar with this code, ignore if I'm just utterly confused_

why are these args hard-coded in place here, instead of being part of the `DEFAULT_ARGS` data structure around L25? digging a little bit, I see this is a pattern throughout, but I find it sort of confusing that there's a set of hardcoded default arguments for each type of tree-builder, and then we're checking for "conflicts with hardcoded defaults" when running each tree-builder, using a different set of arguments.

File: tests/functional/tree.t
Comment (genehack) [inline tests/functional/tree.t:56]: for the sake of symmetry, might be worth adding one additional test that is identical to this plus `--override-default-args` -- or instead having this test duplicate the arguments in the test above _without_ `--override-default-args`.

File: tests/functional/tree.t
Comment (huddlej) [inline tests/functional/tree.t:52]: Is the addition of `--redo` necessary here because we're writing the output of all these tests to `$TMP`? If so, we could just remove references to `$TMP` throughout this file (following [our own cram style guide](https://github.com/nextstrain/augur/blob/master/docs/contribute/DEV_DOCS.md#3-functional-tests)) and skip the `--redo`.

**Review by huddlej — APPROVED** [review]
Thanks, @corneliusroemer! Nice to have the check for modern synonyms and also clearer to use those throughout the docs and tests (`-czb` has always confused me, for example, and `--polytomy` is much clearer).

File: tests/functional/tree.t
Comment (corneliusroemer) [reply tests/functional/tree.t:52]: I'm not quite sure anymore why I needed to add the `--redo` - there was definitely some issue I hit without it - possibly iqtree stubmling upon an existing file from an earlier test without having cleaned up. 

Redo does no harm (it should be the default, and one should actively opt into checkpoints). It might not bring many benefits but given there's no harm I thought I'd sprinkle it.

File: tests/functional/tree.t
Comment (corneliusroemer) [reply tests/functional/tree.t:56]: These cram tests are the slow part of the test suit so I'd probably prefer not to add tests just for symmetry sake - if the tests were fast, then sure!

File: augur/tree.py
Comment (corneliusroemer) [reply augur/tree.py:264]: I _think_ hard-coded args are arguments that are set controlled through `augur tree` options and for which we want to disallow users from overwriting them. Things like: the input alignment, which augur tree processes and sets directly, there's no reason why a user should change that, we want to make this part of the underlying tree binary "private".

Default args are recommendations for tree builder programs that user may override if they want to without causing harm.

Does that make sense?

Comment (genehack) [reply augur/tree.py:264]: > Does that make sense?

It does; albeit the use of "default" to mean two orthogonal things that _look_ like they're different ways of saying the same thing (`DEFAULT_ARGS` versus "hardcoded defaults") seems inherently confusing. 

Maybe "hardcoded defaults" could be changed to "hardcoded options", but this is a nitpick. :shrug:

File: tests/functional/tree.t
Comment (victorlin) [reply tests/functional/tree.t:52]: This would require reorganizing the cram files: https://github.com/nextstrain/augur/issues/1558

I plan to make a PR for that later, but this shouldn't block on it!

Comment (victorlin) [reply tests/functional/tree.t:52]: Actually I'm not sure we can avoid `--redo`. I realized while working on #1559 that IQ-TREE puts intermediate files in the directory of the `--alignment` input file, not the directory of the `--output` file. Only the 2nd is cleaned up after every test.

Comment (corneliusroemer) [reply tests/functional/tree.t:52]: Good discussion here, I've got a fix for this issue in #1667 - because IQtree pollutes the _input_ file directory, it's best to copy the input file data dir to each temporary directory.

We can remove the `-redo` once/if that's merged. For now, I'd prefer to keep it like this to not break something else on the way.

**Pull request merged**
